src/rules/jsxNoBindRule.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "Binds are forbidden in JSX attributes due to their rendering performance impact";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const jsxNoBindWalker = new JsxNoBindWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(jsxNoBindWalker);
+    }
+}
+
+class JsxNoBindWalker extends Lint.RuleWalker {
```

Comment:
would you mind refactoring to use WalkContext and `applyWithFunction`? see https://palantir.github.io/tslint/develop/custom-rules/performance-tips.html and https://palantir.github.io/tslint/develop/custom-rules/walker-design.html. the script in this repo (`./scripts/newRule.js`) ought to scaffold this up for you.